### ISSUE TITLE
Further Improvements to GetItemInfoInstant

### ISSUE THREAD

**Issue by descention** [descention/Compat#5]
Attempts should be made to improve the initial loading of `GetItemInfoInstant`. I'm not sure if there is a solution at this time. Good news is that this doesn't happen repeatedly anymore.

```
1x TradeSkillMaster\CoreAPI\Items-Items.lua:719: Failed to get instant info! (i:6948, classId)
[C]: in function `error'
...rfaceTradeSkillMaster\Debug\ErrorHandler.lua:64: in function `Assert'
TradeSkillMaster\CoreAPI\Items-Items.lua:719: in function <TradeSkillMaster\CoreAPI\Items.lua:708>
(tail call): ?
TradeSkillMaster\CoreAPI\Items-Items.lua:300: in function `IsDisenchantable'
...faceTradeSkillMaster\CoreAPI\Conversions.lua:143: in function `GetValue'
TradeSkillMaster\TradeSkillMaster-v3.6.43.lua:521: in function `LoadTooltip'
TradeSkillMaster\Private\Tooltips.lua:91: in function `callback'
...rfaceTradeSkillMaster\Private\TooltipLib.lua:143: in function <...rfaceTradeSkillMaster\Private\TooltipLib.lua:122>
...rfaceTradeSkillMaster\Private\TooltipLib.lua:83: in function <...rfaceTradeSkillMaster\Private\TooltipLib.lua:82>
[C]: ?
...rfaceTradeSkillMaster\Private\TooltipLib.lua:95: in function `SetBagItem'
FrameXML\ContainerFrame.lua:845: in function `ContainerFrameItemButton_OnEnter'
[string "*:OnEnter"]:1: in function <[string "*:OnEnter"]:1>

Locals:
self = <table> {
 MoveItems = <function> defined @TradeSkillMaster\CoreAPI\Mover.lua:612
 GetSounds = <function> defined @TradeSkillMaster\CoreAPI\Sounds.lua:62
 HasModule = <function> defined @TradeSkillMaster\CoreAPI\Modules.lua:158
 NewModule = <function> defined @TradeSkillMaster\CoreAPI\Modules.lua:44
 GetConnectedRealms = <function> defined @TradeSkillMaster\TradeSkillMaster.lua:832
 Settings = <table> {
 }
 ItemFilter = <table> {
 }
 GetItemValue = <function> defined @TradeSkillMaster\CoreAPI\Prices.lua:95
 GetRegion = <function> defined @TradeSkillMaster\TradeSkillMaster.lua:849
 Auction = <table> {
 }
 Sync = <table> {
 }
 Delay = <table> {
 }
 Inventory = <table> {
 }
 GetNoSoundKey = <function> defined @TradeSkillMaster\CoreAPI\Sounds.lua:58
 Groups = <table> {
 }
 AppHelper = <table> {
 }
 MoneyToString = <function> defined @TradeSkillMaster\CoreAPI\Money.lua:26
 DoPlaySound = <function> defined @TradeSkillMaster\CoreAPI\Sounds.lua:66
 Operations = <table> {
 }
 GetPriceSources = <function> defined @TradeSkillMaster\CoreAPI\Prices.lua:124
 Conversions = <table> {
 }
 Debug = <table> {
 }
 Threading = <table> {
 }
 ModuleAPI = <function> defined @TradeSkillMaster\CoreAPI\Modules.lua:162
 Assert = <function> defined @TradeSkillMaster\Debug\ErrorHandler.lua:60
 GetCustomPriceValue = <function> defined @TradeSkillMaster\CoreAPI\Prices.lua:73
 ValidateCustomPrice = <function> defined @TradeSkillMaster\CoreAPI\Prices.lua:68
 GUI = <table> {
 }
 Player = <table> {
 }
 Design = <table> {
 }
 Item = <table> {
 }
 Util = <table> {
 }
 GetChatFrame = <function> defined @TradeSkillMaster\TradeSkillMaster.lua:820
 MoneyFromString = <function> defined @TradeSkillMaster\CoreAPI\Money.lua:109
 GetTSMProfileIterator = <function> defined @TradeSkillMaster\TradeSkillMaster.lua:806
}
cond = nil
err = "Failed to get instant info! (i:6948, classId)"
raiseLevel = 0
private = <table> {
 AddonBlockedEvent = <function> defined @TradeSkillMaster\Debug\ErrorHandler.lua:292
 origErrorHandler = <function> defined @!BugGrabber\BugGrabber.lua:306
 ErrorHandler = <function> defined @TradeSkillMaster\Debug\ErrorHandler.lua:155
 ignoreErrors = false
 ShowError = <function> defined @TradeSkillMaster\Debug\ErrorHandler.lua:100
 isAssert = true
 isErrorFrameVisible = false
}

```

**Comment by descention** [comment]
>[Andrew](https://discordapp.com/channels/168296152670797824/552993554348113930/1064599026831470592): GetItemInfo reads from the item cache only.
>
>GameTooltip frames will query items if you set them to an uncached item via GameTooltip:SetHyperlink("item:"..itemID). The tooltip does not need to be visible, so you can do like GameTooltip:SetOwner(WorldFrame, "ANCHOR_NONE")
>
>It's not instant, so you'll need some sort of callback system for when it's ready, or to recall whatever is using GetItemInfo on uncached items after a second. 
>
>Should make your own querying tooltip frame so you aren't hijacking GameTooltip from other stuff as well.

**Comment by descention** [comment]
Because addon developers handle bad data well, sending an empty table seems to reduce the number of errors.

**Comment by descention** [comment]
Fixed with a local cache via SavedVariable 1060e470c1d851cbf7c5b78fa16a6287a673b30e
